tests/snappi_tests/multidut/bgp/files/bgp_outbound_helper.py

```diff
@@ -0,0 +1,632 @@
+        flow1.tx_rx.device.rx_names = destination
+        flow1.size.fixed = 1024
+        flow1.rate.percentage = 10
+        flow1.metrics.enable = True
+        flow1.metrics.loss = True
+
+    global total_routes
+    if traffic_type == 'IPv4':
+        for route in route_range['IPv4']:
+            total_routes = total_routes+route[2]
+        createTrafficItem("IPv4_Traffic", [ipv4_src[0]], ipv4_dest)
+    else:
+        for route in route_range['IPv6']:
+            total_routes = total_routes+route[2]
+        createTrafficItem("IPv6 Traffic", [ipv6_src[0]], ipv6_dest)
+    return conv_config
+
+
+def get_flow_stats(cvg_api):
+    """
+    Args:
+        cvg_api (pytest fixture): Snappi API
+    """
+    request = cvg_api.convergence_request()
+    request.metrics.flow_names = []
+    return cvg_api.get_results(request).flow_metric
+
+
+def get_install_time(duthosts,
+                     cvg_api, bgp_config,
+                     flap_test_port,
+                     traffic_type,
+                     iteration):
+    """
+    Args:
+        duthost (pytest fixture): duthost fixture
+        cvg_api (pytest fixture): Snappi API
+        bgp_config: __tgen_bgp_config
+        traffic_type : IPv4 / IPv6 traffic type
+        iteration : Number of iterations
+    """
+    cvg_api.set_config(bgp_config)
+    avg_pld = []
+    delta_frames = 0
+    for i in range(0, iteration):
+        logger.info(
+            '|--------------------------- Iteration : {} -----------------------|'.format(i+1))
+        """ Starting Protocols """
+        logger.info("Starting all protocols ...")
+        cs = cvg_api.convergence_state()
+        cs.protocol.state = cs.protocol.START
+        cvg_api.set_state(cs)
+        wait(TIMEOUT, "For Protocols To start")
+
+        logger.info('Starting Traffic')
+        cs = cvg_api.convergence_state()
+        cs.transmit.state = cs.transmit.START
+        cvg_api.set_state(cs)
+        wait(TIMEOUT, "For Traffic To start")
+
+        flow_stats = get_flow_stats(cvg_api)
+        logger.info('Loss %: {}'.format(int(flow_stats[0].loss)))
+        pytest_assert(int(flow_stats[0].loss) == 0, 'Loss Observed in traffic flow before link Flap')
+        # Flap the required test port
+        if duthosts[0].hostname == flap_test_port['hostname']:
+            logger.info(' Shutting down {} port of {} dut !!'.
+                        format(flap_test_port['port_name'], flap_test_port['hostname']))
+            duthosts[0].command('sudo config interface shutdown {} \n'.
+                                format(flap_test_port['port_name']))
+        else:
+            cs = cvg_api.convergence_state()
+            cs.link.port_names = [flap_test_port['port_name']]
+            cs.link.state = cs.link.DOWN
+            cvg_api.set_state(cs)
+            logger.info('Shutting down snappi port : {}'.format(flap_test_port['port_name']))
+        wait(TIMEOUT, "For link to shutdown")
+        flow_stats = get_flow_stats(cvg_api)
```

Comment:
if the ixia port is shutdown will `get_flow_stats` return accurate `frames_tx_rate`?

Reply:
Here we are shutting down one of the rx port. It should not affect the transmitting side port as both are separate physical ports. So, tx rate will remain same.